src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -113,44 +113,72 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
-                getLogger('channel').error(`Compile Error: ${error}`)
-                throw Error('Failed to compile typescript Lambda')
-            }
+    if (!config.baseBuildDir) {
+        throw Error('invalid state: config.baseBuildDir was not set')
+    }
+    if (config.invokeTarget.target !== 'code') {
+        return
+    }
+
+    async function findTsOrTsConfig(dir: string, child: boolean): Promise<string | undefined> {
+        const glob = (child ? '*/' : '') + '{*.ts,tsconfig.json}'
+        const found = await vscode.workspace.findFiles(new vscode.RelativePattern(dir, glob), '**/node_modules/**', 1)
+        if (found.length === 0) {
+            return undefined
         }
+        return found[0].fsPath
+    }
+
+    // Require tsconfig.json or *.ts in the top-level of the source app, to
+    // indicate a typescript Lambda. #2086
+    // Note: we don't use this tsconfig.json for compiling the target=code
+    // Lambda app below, instead we generate a minimal one.
+    const isTsApp = (await findTsOrTsConfig(config.codeRoot, false)) !== undefined
+    if (!isTsApp) {
+        return
+    }
+
+    const buildOutputDir = path.join(config.baseBuildDir, 'output')
+    const buildDirTsFile = await findTsOrTsConfig(buildOutputDir, true)
+    if (!buildDirTsFile) {
+        // Should never happen: `sam build` should have copied the tsconfig.json from the source app dir.
+        throw new Error(`tsconfig.json or *.ts not found in: "${buildOutputDir}/*"`)
+    }
+    // XXX: `sam` may rename the CodeUri (and thus "output/<app>/" dir) if the
+    // original "<app>/" dir contains special chars, so get it this way. #2086
+    const buildDirApp = path.dirname(buildDirTsFile)
+    const buildDirTsConfig = path.join(buildDirApp, 'tsconfig.json')
+
+    const tsc = await systemutil.SystemUtilities.findTypescriptCompiler()
+    if (!tsc) {
+        throw new Error('TypeScript compiler "tsc" not found in node_modules/ or the system.')
+    }
+
+    // Default config.
+    // Adapted from: https://github.com/aws/aws-toolkit-jetbrains/blob/911c54252d6a4271ee6cacf0ea1023506c4b504a/jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt#L60
+    const defaultTsconfig = {
+        compilerOptions: {
+            target: 'es6',
+            module: 'commonjs',
+            typeRoots: ['node_modules/@types'],
+            types: ['node'],
+            rootDir: '.',
+            inlineSourceMap: true,
+        },
+    }
+    try {
+        // Overwrite the tsconfig.json copied by `sam build`.
+        writeFileSync(buildDirTsConfig, JSON.stringify(defaultTsconfig, undefined, 4))
+        getLogger('channel').info(`Compiling TypeScript with: "${tsc}"`)
+        await new ChildProcess(true, tsc, undefined, '--project', buildDirApp).run()
+    } catch (error) {
+        getLogger('channel').error(`TypeScript compile error: ${error}`)
+        throw Error('Failed to compile TypeScript Lambda')
```

Comment:
Nit: "Lambda" -> "Function" ?

Reply:
that seems ambiguous, since "function" means something very different in typescript generally.

Reply:
renamed to "typescript app"